Compile warnings

When compiling commit 1d4793ead562ccc0195482352e76e7758a908bef, I get the following warnings:

```bash
[40%] Building CXX object libcore/CMakeFiles/core.dir/src/geometry/helper/CorrectGeometry.cpp.o
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/src/geometry/helper/CorrectGeometry.cpp:335:20: warning: loop variable '[_, trainDoor]' of type 'const std::__1::pair<const int, TrainDoor>' creates a copy from type 'const std::__1::pair<const int, TrainDoor>' [-Wrange-loop-analysis]
    for(const auto [_, trainDoor] : train._doors) {
                   ^
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/src/geometry/helper/CorrectGeometry.cpp:335:9: note: use reference type 'const std::__1::pair<const int, TrainDoor> &' to prevent copying
    for(const auto [_, trainDoor] : train._doors) {
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~
                   &
1 warning generated.
```
```bash
[ 95%] Building CXX object libcore/CMakeFiles/core.dir/src/Simulation.cpp.o
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/src/Simulation.cpp:161:24: warning: loop variable '[transID, maxAgents]' of type 'const std::__1::pair<const int, int>' creates a copy from type 'const std::__1::pair<const int, int>' [-Wrange-loop-analysis]
        for(auto const [transID, maxAgents] : groupMaxAgents) {
                       ^
/Users/gjaeger/Documents/hubs/JuPedSim_github/jpscore/libcore/src/Simulation.cpp:161:13: note: use reference type 'const std::__1::pair<const int, int> &' to prevent copying
        for(auto const [transID, maxAgents] : groupMaxAgents) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                       &
[ 96%] Building CXX object libcore/CMakeFiles/core.dir/src/SimulationHelper.cpp.o
1 warning generated.
```

--- a/libcore/src/Simulation.cpp
+++ b/libcore/src/Simulation.cpp
@@ -158,7 +158,7 @@ bool Simulation::InitArgs()
 
         // Read and set max door usage from schedule file
         auto groupMaxAgents = EventFileParser::ParseMaxAgents(_config->GetScheduleFile());
-        for(auto const [transID, maxAgents] : groupMaxAgents) {
+        for(auto const & [transID, maxAgents] : groupMaxAgents) {
             _building->GetTransition(transID)->SetMaxDoorUsage(maxAgents);
         }
     }

--- a/libcore/src/geometry/helper/CorrectGeometry.cpp
+++ b/libcore/src/geometry/helper/CorrectGeometry.cpp
@@ -332,7 +332,7 @@ std::map<int, std::pair<Point, Point>> ComputeTrainDoorCoordinates(
     }
 
     Point start{std::begin(trackWalls)->GetPoint1()};
-    for(const auto [_, trainDoor] : train._doors) {
+    for(const auto & [_, trainDoor] : train._doors) {
         double distanceFromTrackStart = trainStartOffset + trainDoor._distance;
         double width                  = trainDoor._width;
         std::vector<Point> intersectionPoints;

